scheduler/src/cook/config.clj

```diff
@@ -412,6 +412,8 @@
 
      :api-only? (fnk [[:config {api-only? false}]]
                   api-only?)
+     :default-container-for-pool (fnk [[:config {default-container-for-pool {}}]]
+                                   default-container-for-pool)
```

Comment:
A few thoughts:

1. Maybe "image" instead of "container"
1. Maybe this should live under the existing `:pools` section of the config
1. Maybe this should be a regex for the pool name so that we don't have to have an entry for every single pool and remember to add/remove entries when we add/remove pools

Reply:
1. I chose container, because you give the entire container spec, that way we're robust to future image and image schema changes ---- you can specify parameters if you want to. (I borrowed the approach from how the integration tests name it and name the subfield. Its 'container', not 'image'.
2. I like that idea.
3. Regexp could be problematic. What if multiple match? I'm also thinking of test scenario's where we want to use a default everywhere except for a test or snowflake pools and use a different container for that one. (E.g., imaging a GPU pool with a special blessed GPU image that's different from the default everywhere else.)

Reply:
Ah, makes sense on "container". 

For regex, I think you'd want a list of (regex, container) pairs, and you take the first one that matches (order matters). That would allow for the flexibility you're describing without having to have an entry for every single pool and remember to add/remove entries when we add/remove pools.

Reply:
Ok.